Mathlib/FieldTheory/RatFunc/AsPolynomial.lean

```diff
@@ -193,3 +198,47 @@ theorem eval_mul {x y : RatFunc K} (hx : Polynomial.eval₂ f a (denom x) ≠ 0)
 end Field
 
 end Eval
+
+end RatFunc
+
+section AdicValuation
+
+variable (K : Type _) [Field K]
+
+namespace Polynomial
+
+open IsDedekindDomain.HeightOneSpectrum
+
+/-- This is the principal ideal generated by `X` in the ring of polynomials over a field K,
+  regarded as an element of the height-one-spectrum. -/
+def idealX : IsDedekindDomain.HeightOneSpectrum K[X] where
+  asIdeal := Ideal.span {Polynomial.X}
+  isPrime := by rw [Ideal.span_singleton_prime]; exacts [Polynomial.prime_X, Polynomial.X_ne_zero]
+  ne_bot  := by rw [ne_eq, Ideal.span_singleton_eq_bot]; exact Polynomial.X_ne_zero
+
+@[simp]
+theorem idealX_span : (idealX K).asIdeal = Ideal.span {Polynomial.X} := rfl
+
+@[simp]
+theorem valuation_X_eq_neg_one :
+    (idealX K).valuation (RatFunc.X : RatFunc K) = Multiplicative.ofAdd (-1 : ℤ) := by
+  rw [← RatFunc.algebraMap_X, valuation_of_algebraMap, intValuation_singleton]
+  · exact Polynomial.X_ne_zero
+  · exact idealX_span K
+
+end Polynomial
+namespace RatFunc
+
+open scoped Classical DiscreteValuation
+
+open Polynomial
+
+instance : Valued (RatFunc K) ℤₘ₀ := Valued.mk' (idealX K).valuation
```

Comment:
Are you sure we want to choose a specific valuation for `K(X)`?

Reply:
I imagine that your point is that in general $K[X]$ has many prime ideals and therefore many valuations, like the $(X-1)$-adic one, or the $f(X)$-adic one for every possible irreducible $f(X)$. Nevertheless, the field $K(X)$ has a "canonical" choice of variable, and we are incline to consider that the standard valuation on it is the $(X)$-adic one: if you want to put the $(X-1)$-adic, say, it seems more reasonable to perform a change of variables $(X-1)\mapsto Y$ and to use the standard valuation on $K(Y)$. So my answer is probably "I am not _sure_ but I think this is the right thing to do". Of course the other option would be to simply have a `def` here and go for localized instances, but I wonder whether it is worth it.

Reply:
Maybe a scoped instance?

Reply:
I wonder if this would really change anything. If all the material about this valuation is here, whenever it is needed one should `open scoped` and then the instance would fire as if it were a global one. I guess that the true question is whether we really want to avoid a `Valued` instance on polynomials/rational functions (and in this case we should simply let this be a `def` and promote it to an instance locally when needed), or if we're happy with the idea that $K[X]$ has many valuations but only one Valued structure.

Reply:
@riccardobrasca @erdOne Have you more ideas on this `instance` problem?

Reply:
What I was proposing is to keep it a def and add something like
```lean
scoped[foo] attribute [instance] blahblah
```
In this way `open scoped foo` makes it an instance. In particular one can use all the other results without having this instance. But I agree it's not very important.

Reply:
I have no preference either way. I just wanted to make sure that this was a conscious choice, and I'll happily accept what you think is best.

Reply:
> What I was proposing is to keep it a def and add something like
> 
> ```lean
> scoped[foo] attribute [instance] blahblah
> ```
> 
> In this way `open scoped foo` makes it an instance. In particular one can use all the other results without having this instance. But I agree it's not very important.

Yes, this what I meant by "having a `def` and open a local/scoped instance`. Let me do a couple of tests and report back.

Reply:
OK, so here is my take on the situation: there are three ways to go:
1. Leave it a global instance as it is
2. Make it a `def valuedRatFunc` and go Riccardo's way by adding a `scoped [LaurentSeries] attribute [instance] valuedRatFunc` somewhere in the `LaurentSeries` namespace.
3. Make it a `def valuedRatFunc` and simply do an `attribute [local instance] valuedRatFunc` inside the file I am working on.

My first objection to @erdOne 's comment was that going for 2 is "almost equivalent"  to 1: if one would ever need to work with *any* Valued structure on LaurentSeries or Rational Functions, they would need to `open LaurentSeries` (to benefit of all the API) and the instance would fire, and one would have the same problems to declare a `Valued` structure using e.g. the $(X-1)$-adic valuation. In this sense, this is not much of an advantage. One solution would be to have a specific namespace (like $X$-adic valuation, or so) specifying that we are taking *this very* valuation into account: to me it looks a bit too much.

Solution 3 can work, with the obvious drawback that one needs to repeat this bit of code declaring the `local instance` each time one works with this.

If we stick to 1., one can add a bit of doc saying that if one wants to consider other valuations, this `Valued` instance must be locally disabled. This is what happens e.g. with the normed structure on the product of normed spaces if one does want an exotic norm on $\mathbb{R}^N$ which is not the $\ell^2$-norm.

As far as my project with Maria Ines is concerned (and this series of PR's) all options are not hard to implement, so I am happy to take any of the above options. I have a slight preference for 1. (plus the doc warning) but if you see otherwise, I'm all ears.

Reply:
OK, let's go with 1.

bors d+